File: trunk/quests/ikkinz/Diabolic_Destroyer.pl
sub EVENT_SPAWN {
  quest::setnexthpevent(75);
}

sub EVENT_HP {
  if($hpevent == 75) {
    quest::spawn2(294137,0,0,$x,$y-10,$z,$h);
    quest::spawn2(294137,0,0,$x,$y+10,$z,$h);
    quest::setnexthpevent(50);
  }
  if($hpevent == 50) {
    quest::spawn2(294137,0,0,$x,$y-10,$z,$h);
    quest::spawn2(294137,0,0,$x,$y+10,$z,$h);
    quest::setnexthpevent(25);
  }
  if($hpevent == 25) {
    quest::spawn2(294137,0,0,$x,$y-10,$z,$h);
    quest::spawn2(294137,0,0,$x,$y+10,$z,$h);
  }
}

sub EVENT_AGGRO {
  quest::settimer(1,1);
  quest::spawn2(294137,0,0,450,-119,6,200);
  quest::spawn2(294137,0,0,457,-139,6,193);
  quest::spawn2(294137,0,0,438,-158,6,192);
}

sub EVENT_TIMER {
  if(($x > 500) || ($x < 200) || ($y > -30) || ($y < -250)) {
    $npc->GMMove(442,-141,11,192);
  }
}

sub EVENT_DEATH {
  quest::depopall(294137);
  quest::spawn2(294136,0,0,442,-141,11,192);
}

File: trunk/quests/ikkinz/Diabolic_Destroyer.lua
-- Ikkinz Group 1 Expedition
-- Diabolic_Destroyer NPCID: 294043

function event_spawn(e)
	set_event_start();
end

function event_hp(e)
	local xloc = e.self:GetX();
	local yloc = e.self:GetY();
	local zloc = e.self:GetZ();
	local heading = e.self:GetHeading();
	-- spawn #Dire_Illusion x2
	eq.spawn2(294137,0,0,xloc,yloc + 25,zloc,heading);
	eq.spawn2(294137,0,0,xloc,yloc - 25,zloc,heading);
	-- set next HP event threshold
	if (e.hp_event == 76) then
		eq.set_next_hp_event(51);
	elseif (e.hp_event == 51) then
		eq.set_next_hp_event(26);
	end
end

function event_combat(e)
	if (e.joined) then
	-- /loc updated from MySEQ capture 2014/01/26
	eq.spawn2(294137,0,0,418.5914,-179.3205,10.502,192);
	eq.spawn2(294137,0,0,418.5914,-94.3507,10.502,192);
	eq.spawn2(294137,0,0,379.2154,-186.3592,1.502002,256);
	else
		if (e.self:GetHP() < e.self:GetMaxHP()) then
			e.self:SetHP(e.self:GetMaxHP());
		end
		set_event_start();
	end
 end
 
function event_death_complete(e)
	-- depop the Dire_Illusions
	eq.depop_all(294137);
	-- spawn the Cruel_Illusion
	eq.spawn2(294136,0,0,442,-141,11,192);
end

function set_event_start()
	eq.depop_all(294137);
	eq.set_next_hp_event(76);
end